code/datums/components/footstep.dm

```diff
@@ -89,7 +89,7 @@
 			return
 		if(ishuman(LM)) //for proper humans, they're special
 			var/mob/living/carbon/human/H = LM
-			var/feetCover = (H.wear_suit && (H.wear_suit.body_parts_covered & FEET)) || (H.w_uniform && (H.w_uniform.body_parts_covered & FEET) || (H.shoes && (H.shoes.body_parts_covered & FEET)))
+			var/feetCover = HAS_TRAIT(H, TRAIT_HARDSOLES) || (H.wear_suit && (H.wear_suit.body_parts_covered & FEET)) || (H.w_uniform && (H.w_uniform.body_parts_covered & FEET) || (H.shoes && (H.shoes.body_parts_covered & FEET)))
```

Comment:
why would this be here..? This makes it so the footstep noise won't play if you have the trait, even if you're barefoot and it should play the barefoot noise. Are feet shoes?

Reply:
oof, sorry, was just looking for one specific piece of code and kinda replacing it mindlessly